anker/src/processor.rs

```diff
@@ -167,13 +167,112 @@ fn process_deposit(
     Ok(())
 }
 
+/// Return some bSOL and get back the underlying stSOL.
+fn process_withdraw(
+    program_id: &Pubkey,
+    accounts_raw: &[AccountInfo],
+    amount: BLamports,
+) -> ProgramResult {
+    let accounts = WithdrawAccountsInfo::try_from_slice(accounts_raw)?;
+
+    let (solido, anker) = deserialize_anker(
+        program_id,
+        accounts.anker,
+        accounts.solido,
+        accounts.reserve_account,
+    )?;
+
+    anker.check_mint(accounts.b_sol_mint)?;
```

Comment:
Yes, that is true, but we do deserialize the mint before we call `burn_b_sol`, so I think it’s good to check anyway. I think in this case it is fine to deserialize and check the mint later, but in general I would rather err on the side of checking too often, than missing a check. Especially because the mint’s supply is used to compute the exchange rate, which is used to determine how much to burn.